src/Analyzers/Core/CodeFixes/AddExplicitCast/AbstractAddExplicitCastCodeFixProvider.cs

```diff
@@ -67,99 +77,108 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        }
     }
 
-    private SyntaxNode ApplyFix(
+    private (SyntaxNode finalTarget, SyntaxNode finalReplacement) ApplyFix(
         Document document,
         SemanticModel semanticModel,
-        SyntaxNode currentRoot,
         TExpressionSyntax targetNode,
         ITypeSymbol conversionType,
         CancellationToken cancellationToken)
     {
         var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
         var semanticFacts = document.GetRequiredLanguageService<ISemanticFactsService>();
 
-        // if the node we're about to cast already has a cast, replace that cast if both are reference-identity downcasts.
-        if (syntaxFacts.IsCastExpression(targetNode) || syntaxFacts.IsConversionExpression(targetNode))
+        var (currentTarget, currentReplacement) = ApplyFixWorker();
+
+        // If the original target was surrounded by parentheses, we can consider trying to remove that as well.
+        if (syntaxFacts.IsParenthesizedExpression(currentTarget.Parent))
         {
-            GetPartsOfCastOrConversionExpression(targetNode, out var castTypeNode, out var castedExpression);
+            return (currentTarget.Parent, currentTarget.Parent.ReplaceNode(currentTarget, currentReplacement).WithAdditionalAnnotations(Simplifier.Annotation));
+        }
 
-            var castType = semanticModel.GetTypeInfo(castTypeNode, cancellationToken).Type;
-            if (castType != null)
+        return (currentTarget, currentReplacement);
+
+        (SyntaxNode finalTarget, SyntaxNode finalReplacement) ApplyFixWorker()
+        {
+            // if the node we're about to cast already has a cast, replace that cast if both are reference-identity downcasts.
+            if (syntaxFacts.IsCastExpression(targetNode) || syntaxFacts.IsConversionExpression(targetNode))
             {
-                var firstConversion = semanticFacts.ClassifyConversion(semanticModel, castedExpression, castType);
-                var secondConversion = semanticModel.Compilation.ClassifyCommonConversion(castType, conversionType);
+                GetPartsOfCastOrConversionExpression(targetNode, out var castTypeNode, out var castedExpression);
 
-                if (firstConversion is { IsImplicit: false, IsReference: true } or { IsIdentity: true } &&
-                    secondConversion is { IsImplicit: false, IsReference: true })
+                var castType = semanticModel.GetTypeInfo(castTypeNode, cancellationToken).Type;
+                if (castType != null)
                 {
-                    return currentRoot.ReplaceNode(
-                        targetNode,
-                        this.Cast((TExpressionSyntax)castedExpression, conversionType)
-                            .WithTriviaFrom(targetNode)
-                            .WithAdditionalAnnotations(Simplifier.Annotation));
+                    var firstConversion = semanticFacts.ClassifyConversion(semanticModel, castedExpression, castType);
```

Comment:
nit: could move the def of semanticFacts to here

Reply:
yup.  though i'm ok with this.